docs/build-protocol-info.md

```diff
@@ -0,0 +1,190 @@
+100 DOTs for `Reserve` purposes. The account could not make a transfer that brings its free balance
+below 150 DOTs, but an operation could result in reserving DOTs such that the free balance is below
+150, but above 100 DOTs.
+
+Bonding tokens for staking and voting in governance referenda both utilize locks.
+
+Vesting is another abstraction that uses locks on free balance. Vesting sets a lock that decreases
+over time until all the funds are spendable.
+
+More info:
+
+- [Lockable Currency](https://substrate.dev/rustdocs/master/frame_support/traits/trait.LockableCurrency.html)
+- [Lock Withdraw Reasons](https://substrate.dev/rustdocs/master/frame_support/traits/enum.WithdrawReason.html)
+- [Vesting](https://substrate.dev/rustdocs/master/pallet_vesting/index.html)
+
+## Extrinsics and Events
+
+### Extrinsics
+
+Extrinsics constitute information from the outside world and take on three forms:
+
+- Inherents
+- Signed Transactions
+- Unsigned Transactions
+
+As an infrastructure provider, you will deal almost exclusively with signed transactions. You will,
+however, see other extrinsics within the blocks that you decode. Find more information in the
+[Substrate documentation](https://substrate.dev/docs/en/next/conceptual/node/extrinsics).
+
+Inherents contain information that is not provably true, but validators agree on based on some
+measure of reasonability. For example, a timestamp cannot be proved, but validators can agree that
+it is within some delta of their system clock. Inherents are not gossiped on the network, and only
+block authors insert them into blocks.
+
+Signed transactions contain a signature of the account that issued the transaction and stands to
+pay a fee to have the transaction included on chain. Because the value of including signed
+transactions on-chain can be recognized prior to execution, they can be gossiped on the network
+between nodes with a low risk of spam. Signed transactions fit the concept of a transaction in
+Ethereum or Bitcoin.
+
+Some transactions cannot be signed by a fee-paying account and use unsigned transactions. For
+example, when a user claims their DOTs from the Ethereum DOT indicator contract to a new DOT
+address, the new address doesn't have any fees.
+
+### Transaction Mortality
+
+Extrinsics can be mortal or immortal. The transaction payload includes a block number and block
+hash checkpoint from which a transaction is valid and a validity period (also called "era" in some
+places) that represents the number of blocks after the checkpoint for which the transaction is
+valid. If the extrinsic is not included in a block within this validity window, it will be
+discarded from the transaction queue.
+
+Setting the block checkpoint to zero, using the genesis hash, and a validity period of zero will
+make the transaction "immortal".
+
+**NOTE:** If an account is reaped and a user re-funds the account, then they could replay an
+immortal transaction. Always default to using a mortal extrinsic.
```

Comment:
This raises the question - when would one use an immortal extrinsic?

Reply:
It should definitely be rare, but I would say never in a custody situation where some party holds the keys on behalf of some user. But one example could be if you really don't want to pay a tip, but just want a tx to go in whenever activity dies down enough to include it.